provisioning/baremetal_pod.go

```diff
@@ -204,18 +209,29 @@ func setIronicExternalIp(name string, config *metal3iov1alpha1.ProvisioningSpec)
 	}
 }
 
-func setIronicExternalUrl(info *ProvisioningInfo) (corev1.EnvVar, error) {
-	ironicIPs, err := GetRealIronicIPs(info)
-	if err != nil {
-		return corev1.EnvVar{}, fmt.Errorf("failed to get Ironic IP when setting external url: %w", err)
-	}
-
+func setIronicExternalIPv6(info *ProvisioningInfo) (corev1.EnvVar, error) {
 	var ironicIPv6 string
 
-	for _, ironicIP := range ironicIPs {
-		if utilnet.IsIPv6String(ironicIP) {
-			ironicIPv6 = ironicIP
-			break
+	externalIPs, ironicIPs, err := GetImageServerIPs(info)
```

Comment:
See below: you can simplify the output of GetImageServerIPs, and that will simplify this function as well.